src/components/form/fields/schema-field.ts

```diff
@@ -0,0 +1,76 @@
+import { LimeElementsAdapter } from '../adapters';
+import JSONSchemaField from 'react-jsonschema-form/lib/components/fields/SchemaField';
+import React from 'react';
+import { FieldProps } from './types';
+
+const hasCustomComponent = (schema): boolean => {
+    return Boolean(schema.lime?.component?.name);
+};
+
+const getCustomComponent = (
+    schema
+): { name: string; props: { [key: string]: any } } => {
+    const name = schema.lime?.component?.name;
+    const props = schema.lime?.component?.props || {};
+
+    return { name: name, props: props };
+};
+
+const buildCustomComponentProps = (props: FieldProps) => {
+    const {
+        formData,
+        disabled,
+        required,
+        readonly,
+        name,
+        registry,
+        schema,
+        errorSchema,
+    } = props;
+
+    return {
+        value: formData,
+        required: required,
+        readonly: readonly,
+        name: name,
+        disabled: disabled,
+        title: schema.title,
+        helperText: schema.description, // TODO: pull error from errorSchema
+        formInfo: {
+            schema: schema,
+            rootSchema: registry.formContext.schema,
+            errorSchema: errorSchema,
```

Comment:
im not 100% what it looks like yet i haven't gotten it to populate in testing yet. The documentation for it can be found here: https://react-jsonschema-form.readthedocs.io/en/latest/advanced-customization/custom-widgets-fields/#field-props